benchmark-overhead/src/test/java/io/opentelemetry/results/CsvPersister.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.results;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Consumer;
+
+class CsvPersister implements ResultsPersister {
+
+  private final Path resultsFile;
+
+  public CsvPersister(Path resultsFile) {this.resultsFile = resultsFile;}
+
+  @Override
+  public void write(List<AppPerfResults> results) {
+
+    ensureFileCreated(results);
+
+    StringBuilder sb = new StringBuilder().append(System.currentTimeMillis() / 1000);
+    // Don't be confused by the loop -- This generates a single long csv line.
+    // Each result is for a given agent run, and we want all the fields for all agents on the same
+    // line so that we can create a columnar structure that allows us to more easily compare agent
+    // to agent for a given run.
+    doSorted(results, result -> {
+      sb.append(",").append(result.startupDurationMs);
+      sb.append(",").append(result.heapUsed.min);
+      sb.append(",").append(result.heapUsed.max);
+      sb.append(",").append(result.getTotalAllocatedMB());
+      sb.append(",").append(result.totalGCTime);
+      sb.append(",").append(result.maxThreadContextSwitchRate);
+      sb.append(",").append(result.iterationAvg);
+      sb.append(",").append(result.iterationP95);
+      sb.append(",").append(result.requestAvg);
+      sb.append(",").append(result.requestP95);
+      sb.append(",").append(result.peakThreadCount);
+    });
+    sb.append("\n");
+    try {
+      Files.writeString(resultsFile, sb.toString(), StandardOpenOption.APPEND);
```

Comment:
Instead of opening the result file twice and aggregating the results in string builders, why not use the `Writer` for that?

Reply:
I like that it consolidates the mandatory exception handling to a single line instead of spreading it across multiple...but there is more than one way to skin a 🐱 .